Add zed-netcoredbg to csharp extension, add runnables

Tiggilyboo opened the pull request:
This PR adds debugging support #12 as originally implemented by https://github.com/qwadrox/zed-netcoredbg I merged this code  over for debug use. It uses netcoredbg: https://github.com/Samsung/netcoredbg

My PR adds runnables which detect test files, so test fixtures and individual tests can be run or debugged.
<img width="1669" height="1233" alt="netcoredbg_csharp_test_running" src="https://github.com/user-attachments/assets/6f3d87b1-6e94-4044-bf92-9f4b39f6f14c" />

There is however one caveat, I can't seem to find a way to properly attach to the running test pid as we cannot inspect the process from within the extension. As seen in the screenshot below, the user can read and manually attach to the process id in the current implementation until a better solution is found. The only other way I can think of at the moment is to run the tests and output a temporary file which can be read by the extension, or somehow get the tcp connection working.

<img width="1669" height="1233" alt="netcoredbg_csharp_test_showing_pid" src="https://github.com/user-attachments/assets/4a3e5563-c1f2-4a06-88b4-d0f7a70a54cf" />

Open to other ideas, but at least this is one step forward to integrated test runnables from the csharp extension...

From my git commit log for more details:

- Use extension from https://github.com/qwadrox/zed-netcoredbg to have basic debug functionality with DAP and download netcoredbg for the running platform.
- Add runnables and tasks to debug from test files. Captures for nunit, xunit, vstest. Not strict, purely functional (does not pair keywords) to test framework.
- Currently unable to fully attach to runnable, but starts the vstest process so that it can be easily attached with PID listed in console

qwadrox commented on src/binary_manager.rs:
```diff
@@ -0,0 +1,283 @@
+    }
+
+    /// Validates that the binary exists
+    fn validate_binary(&self, binary_path: &str) -> Result<(), String> {
```
And I cached before validation...

Tiggilyboo committed:
Add zed-netcoredbg to csharp extension, add runnables

- Use extension from https://github.com/qwadrox/zed-netcoredbg to
    have basic debug functionality with DAP and download netcoredbg for
    the running platform.
- Add runnables and tasks to debug from test files. Captures for nunit,
    xunit, vstest. Not strict, purely functional (does not pair keywords)
    to test framework.
- Currently unable to fully attach to runnable, but starts the vstest
    process so that it can be easily attached with PID listed in console

Tiggilyboo committed:
Merge branch 'main' into feature/netcoredbg-and-runnables

cm4ker commented:
https://github.com/MattParkerDev/sharpdbg/ - I found this. Perhaps it's worth looking here as well.

Tiggilyboo committed:
Merge branch 'main' into feature/netcoredbg-and-runnables

Tiggilyboo committed:
Merge branch 'main' into feature/netcoredbg-and-runnables

GustavEikaas commented on src/csharp.rs:
```diff
@@ -55,6 +99,243 @@ impl zed::Extension for CsharpExtension {
+        }
+    }
+
+    fn dap_locator_create_scenario(
```
Total Zed and Rust noob here, but reading through this PR it looks like this currently handles the VSTest path only?

Just wanted to provide some context in case it's useful for future work — not criticism of the approach here, this PR is awesome.

There are currently two test platforms in .NET:

* VSTest
* MTP (Microsoft Testing Platform)

If support for MTP is ever added, I think there are a couple of additional pieces that would be needed:

1. Detect whether the project is using MTP or VSTest. One way to do that is:

```bash
dotnet msbuild -getProperty:IsTestingPlatformApplication
```

If stdout is `true` then it's an MTP project.

2. For MTP, it should be possible to build the project, resolve the test assembly path via `TargetPath`, and issue a DAP `launch` request against the test executable/assembly rather than using the VSTest attach flow.

For example:

```bash
dotnet msbuild -getProperty:TargetPath
```

which returns the path to the built test assembly (assuming the active configuration, typically `Debug`).

3. Test filtering would need different handling since the filter syntax differs from VSTest.

One interesting thing about MTP is that it also supports a server/RPC mode, which could potentially allow test execution and result collection without relying on output parsing:

https://github.com/GustavEikaas/easy-dotnet-server/blob/main/EasyDotnet.IDE/TestRunner/Adapters/MTP/RPC/MtpClient.cs#L59

Not suggesting any of this belongs in this PR, but I thought it might be useful context for anyone looking at test debugging support longer-term.
